File: Data Structures & Algorithms/max-water-container/submission-4.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        l, r = 0, (len(heights) - 1)
        max_water = 0
        while l < r:
            if heights[l] < heights[r]:
                max_water = max(max_water, heights[l] * (r - l))
                l += 1
            else:
                max_water = max(max_water, heights[r] * (r - l))
                r -= 1
        return max_water

        

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-4.java
class Solution {
    public int maxProfit(int[] prices) {
        if (prices == null || prices.length < 2){
            return 0;
        }

        int left = 0;
        int right = 1;
        int maxProfit = 0;

        while (right < prices.length){
            if (prices[right] > prices[left]){
                maxProfit = Math.max(maxProfit, prices[right] - prices[left]);
            }else {
                left = right;
            }
            right++;
        }
        return maxProfit;
    }
}
